src/material-examples/cdk-datepicker-overview/cdk-datepicker-overview-example.ts

```diff
@@ -0,0 +1,50 @@
+import {Component} from '@angular/core';
+import {CalendarView, DateAdapter} from '@angular/cdk/datetime';
+
+
+/** @title Basic CDK datepicker */
+@Component({
+  selector: 'cdk-datepicker-overview-example',
+  templateUrl: 'cdk-datepicker-overview-example.html',
+  styleUrls: ['cdk-datepicker-overview-example.css'],
+})
+export class CdkDatepickerOverviewExample {
+  messages: string[] = [];
+  _dateSelected() {
+    this.messages.push('Date has changed. ');
+  }
+}
+
+
+@Component( {
+  selector: 'my-calendar',
+  outputs: ['selectedChange'],
+  template: `
+    <div *ngFor="let date of dates">
+      <button (click)="_selected(date)">{{date}}}</button>
+    </div>
+    <div>Date: {{selected}}</div>
+  `,
+  providers: [{provide: CalendarView, useExisting: MyCalendar}],
+})
+export class MyCalendar<D> extends CalendarView<D> {
+  dates: D[] = [];
```

Comment:
shouldn't this be an `@Input` so the user can pass dates in?

Reply:
Whoops! I forgot